### PR TITLE
Add cluster ID to error

### PR DESCRIPTION
**Pull request by rmweir**
Make error more useful by including cluster ID.

### CODE REVIEW

**Review by jiaqiluo — DISMISSED** [review]
LGTM. We should backport this PR to 2.5 if it is applicable.

File: pkg/controllers/managementuser/settings/controller.go [hunk 1]
```diff
@@ -42,7 +45,7 @@ func (h *handler) onChange(key string, obj *v3.Setting) (runtime.Object, error)
 		return nil, nil
 	}
 
-	return obj, h.apply.
+	err := h.apply.
```
Comment (dramich) [inline pkg/controllers/managementuser/settings/controller.go:48]: This apply should use a cache so we don't have to hit the downstream API i.e. `WithCacheTypes`

**Review by snasovich — APPROVED** [review]

File: pkg/controllers/managementuser/settings/controller.go [hunk 2]
```diff
@@ -42,13 +46,18 @@ func (h *handler) onChange(key string, obj *v3.Setting) (runtime.Object, error)
...
 		WithOwner(obj).
+		WithCacheTypes().
```
Comment (dramich) [inline pkg/controllers/managementuser/settings/controller.go:51]: It requires a cache to be passed in

Comment (dramich) [reply pkg/controllers/managementuser/settings/controller.go:51]: https://github.com/rancher/rancher/blob/release/v2.6/pkg/controllers/management/authprovisioningv2/authprovisioningv2.go#L68

Comment (rmweir) [reply pkg/controllers/managementuser/settings/controller.go]: Forgot to remove this. I did this above in Register() where I have access to the cache.

**Review by snasovich — APPROVED** [review]

**Review by dramich — APPROVED** [review]

**Review by jiaqiluo — APPROVED** [review]

**Comment by nickgerace** [comment]
https://github.com/rancher/rancher/issues/35045

**Pull request merged**